bser/encode.go

```diff
@@ -120,6 +120,9 @@ func encode(buf []byte, d interface{}) ([]byte, error) {
 			order.PutUint64(b, math.Float64bits(r.Float()))
 			return appendItem(buf, 0x07, b), nil
 		case reflect.Slice, reflect.Array:
+			if r.Type().Elem().Kind() == reflect.Struct {
```

Comment:
add a todo to handle pointers to structs

Reply:
@jonasi I'll need to double check but won't that already be handled? Looking at lines 103-111 it looks like for pointers we just call `encode()` on the dereferenced value which will be handled by this block if applicable.

```
elem := false
for r.Kind() == reflect.Ptr || r.Kind() == reflect.Interface {
	r = r.Elem()
	elem = true
}

if elem {
	return encode(buf, r.Interface())
}
```